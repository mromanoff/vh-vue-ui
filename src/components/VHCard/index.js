import VHCard from './VHCard.vue';

export default VHCard;
