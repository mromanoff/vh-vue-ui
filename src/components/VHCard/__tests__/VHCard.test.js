import { shallowMount } from '@vue/test-utils';
import VHCard from '../VHCard.vue';

describe('VHCard.vue', () => {
  let wrapper;

  it('renders default VHCard', () => {
    wrapper = shallowMount(VHCard);
    expect(wrapper.props().block).toBe(false);
    expect(wrapper.props().color).toBe(null);
    expect(wrapper.props().horizontal).toBe(false);
    expect(wrapper.props().hover).toBe(false);
    expect(wrapper.props().nuxt).toBe(false);
    expect(wrapper.props().padded).toBe(false);
    expect(wrapper).toMatchSnapshot();
  });

  it('renders red header background', () => {
    wrapper = shallowMount(VHCard, {
      propsData: {
        color: 'red',
      },
    });
    expect(wrapper.props().block).toBe(false);
    expect(wrapper.props().color).toBe('red');
    expect(wrapper.props().horizontal).toBe(false);
    expect(wrapper.props().hover).toBe(false);
    expect(wrapper.props().nuxt).toBe(false);
    expect(wrapper.props().padded).toBe(false);
    expect(wrapper).toMatchSnapshot();
  });

  it('renders blue header background', () => {
    wrapper = shallowMount(VHCard, {
      propsData: {
        color: 'blue',
        block: true,
      },
    });
    expect(wrapper.props().block).not.toBe(false);
    expect(wrapper.props().color).toBe('blue');
    expect(wrapper.props().horizontal).toBe(false);
    expect(wrapper.props().hover).toBe(false);
    expect(wrapper.props().nuxt).toBe(false);
    expect(wrapper.props().padded).toBe(false);
    expect(wrapper).toMatchSnapshot();
  });
});
