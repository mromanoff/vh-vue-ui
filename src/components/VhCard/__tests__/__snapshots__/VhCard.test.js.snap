// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`VhCard.vue renders blue header background 1`] = `
<div class="VhCard VhCard--block VhCard--blue">
  <div class="VhCard-default"></div>
  <div class="VhCard-header"></div>
  <div class="VhCard-content">
    <div class="VhCard-title"></div>
  </div>
  <div class="VhCard-action"></div>
</div>
`;

exports[`VhCard.vue renders default VhCard 1`] = `
<div class="VhCard">
  <div class="VhCard-default"></div>
  <div class="VhCard-header"></div>
  <div class="VhCard-content">
    <div class="VhCard-title"></div>
  </div>
  <div class="VhCard-action"></div>
</div>
`;

exports[`VhCard.vue renders red header background 1`] = `
<div class="VhCard VhCard--red">
  <div class="VhCard-default"></div>
  <div class="VhCard-header"></div>
  <div class="VhCard-content">
    <div class="VhCard-title"></div>
  </div>
  <div class="VhCard-action"></div>
</div>
`;
